### PR TITLE
DAOS-10339 control: Fix pool creation drpc response (#8864)

### PR DESCRIPTION
**Pull request by knard-intel**
# Description

Fix fields of the pool creation drpc reponse. Some of these fields where not filled by the engine: the control plane was using value from the request which could be different from what was really done.

### CODE REVIEW

**Comment by knard-intel** [comment]
Two errors in functional which seems to not be related to this PR.
Relaunching Jenkins.

**Comment by knard-intel** [comment]
From my understanding the errors are not related to this PR: errors for starting systemd DAOS services => probably a transient errors with the CI nodes.

**Review by tanabarr — APPROVED** [review]

**Review by liw — APPROVED** [review]

File: src/tests/ftest/pool/create_query.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    def test_create_and_query(self):
+        """JIRA ID: DAOS-10339
+
+        Test Description:
+            Create a pool and check that the size of the response is close enough to the requested
+            size.  Verify that the size of pool is equal to the size returned in the response of the
+            pool create.
+
+        :avocado: tags=all,pr,daily_regression
```
Comment (shimizukko) [inline src/tests/ftest/pool/create_query.py:28]: Do we need both pr and daily_regression?

**Review by shimizukko — DISMISSED** [review]

**Review by mjmac — APPROVED** [review]

**Comment by mjmac** [comment]
I recommend using just `Features: pool` on this one. We need to land Kris' fix for the Notice-level logging on release/2.2 in order for `Features: control` to pass.

In any case, it seems pretty suspect that the test you added for this PR failed... Not sure a gatekeeper will sign off on force-landing with that failure unresolved. The error message is strange:
```
avocado.core.exceptions.TestSetupFail: Failed to stop servers:
  Error stopping the systemd subprocess: Error(s) detected gathering daos_server.service log data on server-A,server-B,server-C,server-D
  Error resetting NVMe storage
```

Almost like there was some problem with the yaml parsing. Maybe this is related to @phender's recent big change related to how tests specify the servers?

**Review by mjmac — CHANGES_REQUESTED** [review]
Oops, meant to request changes, not approve.

File: src/tests/ftest/pool/create_query.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        :avocado: tags=hw,medium,ib2
```
Comment (daltonbohning) [inline src/tests/ftest/pool/create_query.py:29]: Just FYI - ib2 is not used anymore

Comment (knard-intel) [reply src/tests/ftest/pool/create_query.py:29]: Thanks for the tip :)

**Comment by knard-intel** [comment]
> I recommend using just `Features: pool` on this one. We need to land Kris' fix for the Notice-level logging on release/2.2 in order for `Features: control` to pass.
> 
> In any case, it seems pretty suspect that the test you added for this PR failed... Not sure a gatekeeper will sign off on force-landing with that failure unresolved. The error message is strange:
> 
> ```
> avocado.core.exceptions.TestSetupFail: Failed to stop servers:
>   Error stopping the systemd subprocess: Error(s) detected gathering daos_server.service log data on server-A,server-B,server-C,server-D
>   Error resetting NVMe storage
> ```
> 
> Almost like there was some problem with the yaml parsing. Maybe this is related to @phender's recent big change related to how tests specify the servers?

Going to investigate on this according to @phender latest update of yaml file configuration.

**Comment by knard-intel** [comment]
> I recommend using just `Features: pool` on this one. We need to land Kris' fix for the Notice-level logging on release/2.2 in order for `Features: control` to pass.
> 
> In any case, it seems pretty suspect that the test you added for this PR failed... Not sure a gatekeeper will sign off on force-landing with that failure unresolved. The error message is strange:
> 
> ```
> avocado.core.exceptions.TestSetupFail: Failed to stop servers:
>   Error stopping the systemd subprocess: Error(s) detected gathering daos_server.service log data on server-A,server-B,server-C,server-D
>   Error resetting NVMe storage
> ```
> 
> Almost like there was some problem with the yaml parsing. Maybe this is related to @phender's recent big change related to how tests specify the servers?

Thanks for your remarks, it was indeed an error from my side.

File: src/tests/ftest/pool/create_query.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/python
```
Comment (daltonbohning) [inline src/tests/ftest/pool/create_query.py:1]: FYI - we shouldn't add the shebang for files that aren't directly executable. But no harm done
```suggestion
```

**Review by mjmac — APPROVED** [review]

**Review by daltonbohning — COMMENTED** [review]
ftest LGTM

File: src/tests/ftest/pool/create_query.py (hunk 3 above)
Comment (knard-intel) [reply src/tests/ftest/pool/create_query.py:1]: I was not aware of that.
Thanks for the info.

Comment (daltonbohning) [reply src/tests/ftest/pool/create_query.py:1]: It's a new change :) In short, if a file doesn't have `execute` permission, it probably shouldn't have a shebang

**Pull request merged**